cmd/zetatool/inbound_ballot/bitcoin.go

```diff
@@ -0,0 +1,215 @@
+	"github.com/btcsuite/btcd/rpcclient"
+	"github.com/rs/zerolog"
+
+	"github.com/zeta-chain/node/cmd/zetatool/config"
+	"github.com/zeta-chain/node/pkg/chains"
+	"github.com/zeta-chain/node/pkg/coin"
+	"github.com/zeta-chain/node/pkg/rpc"
+	crosschaintypes "github.com/zeta-chain/node/x/crosschain/types"
+	"github.com/zeta-chain/node/x/observer/types"
+	"github.com/zeta-chain/node/zetaclient/chains/bitcoin/common"
+	zetaclientObserver "github.com/zeta-chain/node/zetaclient/chains/bitcoin/observer"
+)
+
+func BtcInboundBallotIdentified(ctx context.Context,
+	cfg config.Config,
+	zetacoreClient rpc.Clients,
+	inboundHash string,
+	inboundChain chains.Chain,
+	zetaChainID int64) (string, error) {
+	params, err := chains.BitcoinNetParamsFromChainID(inboundChain.ChainId)
+	if err != nil {
+		return "", fmt.Errorf("unable to get bitcoin net params from chain id: %s", err)
+	}
+
+	connCfg := &rpcclient.ConnConfig{
+		Host:         cfg.BtcHost,
+		User:         cfg.BtcUser,
+		Pass:         cfg.BtcPassword,
+		HTTPPostMode: true,
+		DisableTLS:   true,
+		Params:       params.Name,
+	}
+	rpcClient, err := rpcclient.New(connCfg, nil)
+	if err != nil {
+		return "", fmt.Errorf("error creating rpc client: %s", err)
+	}
+
+	err = rpcClient.Ping()
+	if err != nil {
+		return "", fmt.Errorf("error ping the bitcoin server: %s", err)
+	}
+	res, err := zetacoreClient.Observer.GetTssAddress(context.Background(), &types.QueryGetTssAddressRequest{})
+	if err != nil {
+		return "", fmt.Errorf("failed to get tss address %s", err.Error())
+	}
+	tssBtcAddress := res.GetBtc()
+
+	return bitcoinBallotIdentifier(
+		ctx,
+		rpcClient,
+		params,
+		tssBtcAddress,
+		inboundHash,
+		inboundChain.ChainId,
+		zetaChainID,
+	)
+}
+
+func bitcoinBallotIdentifier(ctx context.Context,
+	btcClient *rpcclient.Client,
+	params *chaincfg.Params,
+	tss string,
+	txHash string,
+	senderChainID int64,
+	zetacoreChainID int64) (string, error) {
+	hash, err := chainhash.NewHashFromStr(txHash)
+	if err != nil {
+		return "", err
+	}
+
+	tx, err := btcClient.GetRawTransactionVerbose(hash)
+	if err != nil {
+		return "", err
+	}
+
+	blockHash, err := chainhash.NewHashFromStr(tx.BlockHash)
+	if err != nil {
+		return "", err
+	}
+
+	blockVb, err := btcClient.GetBlockVerboseTx(blockHash)
+	if err != nil {
+		return "", err
+	}
+
+	if len(blockVb.Tx) <= 1 {
+		return "", fmt.Errorf("block %d has no transactions", blockVb.Height)
+	}
+	// #nosec G115 always positive
+
+	event, err := zetaclientObserver.GetBtcEvent(
+		btcClient,
+		*tx,
+		tss,
+		uint64(blockVb.Height),
+		zerolog.New(zerolog.Nop()),
+		params,
+		common.CalcDepositorFee,
+	)
+	if err != nil {
+		return "", fmt.Errorf("error getting btc event: %s", err)
+	}
+
+	if event == nil {
+		return "", fmt.Errorf("no event built for btc sent to TSS")
+	}
+
+	return identifierFromBtcEvent(event, senderChainID, zetacoreChainID)
+}
+
+func identifierFromBtcEvent(event *zetaclientObserver.BTCInboundEvent,
```

Comment:
This follows similar logic to as processing Bitcoin Inbound Trackers , we can consider refactoring the tracker logic in the future so that it can be reused .

Similar comments for EVM

Solana logic could be used as it needed only minor changes on the zetaclient side

Reply:
The solution also bypasses som checks such as 
- Checking for restricted address 
- Checking for Donation message (IsEventProcessable check) 

I don't think those are required for a debugging tool, but they can be added if needed

We should consider though to refactor the client logic so derive out the common logic from this `GetBallotTool` , `ProcessInboundTracker` and in some cases `ProcessInbound`